### ISSUE TITLE
Better diff in assertion errors

### ISSUE THREAD

**Issue by rodfersou** [assertpy/assertpy#123]
Found this plugin [pytest-icdiff](https://github.com/hjwp/pytest-icdiff) which do a great job from the pictures to do diff

![image](https://user-images.githubusercontent.com/131209/158355014-47c9343a-1243-4a9c-b924-69a4b67c8bd3.png)

but it is good just for `dicts`, and don't work with `assertpy`

I suggest a similar approach, but using [beeprint](https://github.com/panyanyany/beeprint) instead of [pprintpp](https://github.com/wolever/pprintpp)

by depending on those packages, we can make `assertpy` code more sane to maintain, and focus on the fluent api

**Comment by saturnboy** [comment]
@rodfersou a big part of what makes pytest so awesome is the extensive set of plugins.  And yes, side-by-side diffs look great in the terminal.  A core tenet of assertpy is pure python, no dependencies, so not going to extend it like this.

**Comment by rodfersou** [comment]
so, why not deliver the responsability back to pytest? I want to use the plugin instead of assertpy way

**Comment by rodfersou** [comment]
IMHO assertpy should focus on the fluent api

**Comment by saturnboy** [comment]
good point...happy to look at a PR that turns assertpy into a pytest plugin or alternately just leverages the pytest test result rewriting

**Comment by ExplorerOL** [comment]
Hello! I made a fork of assertpy in which pytest assertions rewritings are used: https://github.com/ExplorerOL/assertpy/tree/feature/pytest-assertions-rewriting
Are you interested is such pull request?
